# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/AudioConduit.h
# Commit: d0b311007c03
# Full Hash: d0b311007c033e83824f5f6996a70ab9e870f31f
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 5
# Description:
#   Bug 1654112 - Get RTCP BYE and RTP timeout handling working again (from Bug 1595479) r=mjf,dminor
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106145
# ==============================================================================

diff -r daa9eac709dc -r d0b311007c03 dom/media/webrtc/libwebrtcglue/AudioConduit.h
--- a/dom/media/webrtc/libwebrtcglue/AudioConduit.h	Fri Feb 19 16:20:38 2021 +0000
+++ b/dom/media/webrtc/libwebrtcglue/AudioConduit.h	Fri Feb 19 15:56:41 2021 -0600
@@ -25,6 +25,7 @@
  *  - media-source and target to external transport
  */
 class WebrtcAudioConduit : public AudioSessionConduit,
+                           public webrtc::RtcpEventObserver,
                            public webrtc::Transport {
  public:
   /**
@@ -34,6 +35,11 @@
   void ReceivedRTPPacket(const uint8_t* data, int len,
                          webrtc::RTPHeader& header) override;
 
+  void OnRtcpBye() override;
+  void OnRtcpTimeout() override;
+
+  void SetRtcpEventObserver(mozilla::RtcpEventObserver* observer) override;
+
   /**
    * APIs used by the registered external transport to this Conduit to
    * feed in received RTCP Frames to the Receiver for decoding.
@@ -270,6 +276,9 @@
 
   // Accessed only on mStsThread
   Maybe<DOMHighResTimeStamp> mLastRtcpReceived;
+
+  // Accessed only on main thread.
+  mozilla::RtcpEventObserver* mRtcpEventObserver = nullptr;
 };
 
 }  // namespace mozilla